compiler/rustc_hir_typeck/src/demand.rs

```diff
@@ -1233,6 +1233,22 @@ impl<'a, 'tcx> FnCtxt<'a, 'tcx> {
                             sugg_sp = receiver.span;
                         }
                     }
+
+                    if let hir::ExprKind::Unary(hir::UnOp::Deref, ref inner) = expr.kind
```

Comment:
it would be nice to generalize this to multiple "&&" but whatever, this seems like a very common case already!